### PR TITLE
Miscellaneous updates

### PR DESCRIPTION
**Pull request by shaynevanasperen**
- Some minor tweaks to code quality analysis and styling/formatting of existing packages.

- Several new packages

### CODE REVIEW

File: src/Code/Diagnostics/Command.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+				process.StandardInput.WriteLine($"{command} {arguments}");
+				process.StandardInput.WriteLine($"exit {exitCode}");
+			}
+
+			var milliseconds = Math.Abs((int)timeout.TotalMilliseconds);
+			if (process.WaitForExit(milliseconds) &&
+				output.WaitHandle.WaitOne(milliseconds) &&
+				error.WaitHandle.WaitOne(milliseconds))
+			{
+				hasError ??= text => false;
```
Comment (slang25) [inline src/Code/Diagnostics/Command.cs]: You might want to be careful about using newer language features, unless you have a mechanism in msbuild to check the language version, this is difficult though as there is no msbuild way of resolving the actual language version, it's all passed into csc to figure out.

Comment (shaynevanasperen) [reply src/Code/Diagnostics/Command.cs]: That's a very good point that I hadn't considered. Well spotted!

File: src/Code/Diagnostics/Command.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+			if (timeout == TimeSpan.Zero) throw new ArgumentException("Timeout must be non-zero.", nameof(timeout));
+			if (logger == null) throw new ArgumentNullException(nameof(logger));
+
+			using (var process = StartProcess(command, arguments, commandType, logger, out var output, out var error))
+			{
+				if (commandType == CommandType.Terminal)
+				{
+					var exitCode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "%errorlevel%" : "$?";
+
+					process.StandardInput.WriteLine($"{command} {arguments}");
```
Comment (shaynevanasperen) [inline src/Code/Diagnostics/Command.cs:48]: Check that this works in this context. Can use `timeout` windows executable to test.

File: src/Code/Diagnostics/Command.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+					if (commandType != CommandType.Shell && process.ExitCode != 0 ||
+						hasError(output.Text.ToString()) ||
+						hasError(error.Text.ToString()))
+					{
+						throw new Exception($"{command} exited with code {process.ExitCode}.{Environment.NewLine}{output.Text}{Environment.NewLine}{error.Text}");
+					}
+				}
+				else
+				{
+					throw new Exception($"{command} timed out.");
```
Comment (shaynevanasperen) [inline src/Code/Diagnostics/Command.cs:67]: Could do with a custom exception type.

File: src/Code/Diagnostics/Command.cs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+				if (process.WaitForExit(milliseconds) &&
+					output.WaitHandle.WaitOne(milliseconds) &&
+					error.WaitHandle.WaitOne(milliseconds))
+				{
+					hasError = hasError ?? (text => false);
+					if (commandType != CommandType.Shell && process.ExitCode != 0 ||
+						hasError(output.Text.ToString()) ||
+						hasError(error.Text.ToString()))
+					{
+						throw new Exception($"{command} exited with code {process.ExitCode}.{Environment.NewLine}{output.Text}{Environment.NewLine}{error.Text}");
```
Comment (shaynevanasperen) [inline src/Code/Diagnostics/Command.cs:62]: Could do with a custom exception type.

**Review by anatoliimv — APPROVED** [review]

File: src/Code/Diagnostics/Command.cs (hunk 3 above)
Comment (shaynevanasperen) [reply src/Code/Diagnostics/Command.cs:67]: Actually, after thinking about this some more, I can't think of a _realistic_ example of where having a custom exception type here would help. If someone else comes up with one later, I'll happily accept a PR to add one.

File: src/Code/Diagnostics/Command.cs (hunk 4 above)
Comment (shaynevanasperen) [reply src/Code/Diagnostics/Command.cs:62]: Actually, after thinking about this some more, I can't think of a _realistic_ example of where having a custom exception type here would help. If someone else comes up with one later, I'll happily accept a PR to add one.

**Pull request merged**